umbra-js/src/classes/Umbra.ts

```diff
@@ -313,107 +313,161 @@ export class Umbra {
-    }
-    const userAnnouncements = await this.userAnnouncementsFromLogs(spendingPublicKey, viewingPrivateKey, startBlock, endBlock); // prettier-ignore
-    return { userAnnouncements };
+    return this.fetchAllAnnouncementFromLogs(startBlock, endBlock);
   }
 
-  // ======================================= HELPER METHODS ========================================
+  /**
+   * @notice Fetches all Umbra event logs using The Graph
+   * @dev Currently ignores the start and end block parameters and returns all events; this may change in a
+   * future version
+   * @param startBlock Ignored
+   * @param endBlock Ignored
+   * @returns A list of Announcement events supplemented with additional metadata, such as the sender, block,
+   * timestamp, txhash, and the subgraph identifier
+   */
+  async fetchAllAnnouncementsFromSubgraph(
+    startBlock: string | number,
+    endBlock: string | number
+  ): Promise<SubgraphAnnouncement[]> {
+    if (!this.chainConfig.subgraphUrl) {
+      throw new Error('Subgraph URL must be defined to fetch via subgraph');
+    }
+
+    // TODO: We're ignoring these overrides for The Graph. Is this intentional? Should we remove the parameters,
+    // or update so it uses them?
+    startBlock;
+    endBlock;
+
+    // Query subgraph
+    const subgraphAnnouncements: SubgraphAnnouncement[] = await recursiveGraphFetch(
+      this.chainConfig.subgraphUrl,
+      'announcementEntities',
+      (filter: string) => `{
+        announcementEntities(${filter}) {
+          amount
+          block
+          ciphertext
+          from
+          id
+          pkx
+          receiver
+          timestamp
+          token
+          txHash
+        }
+      }`
+    );
+
+    return subgraphAnnouncements;
+  }
 
   /**
-   * @notice Queries the node for logs, and returns the set of Announcements that were intended for the specified user
-   * @param spendingPublicKey Receiver's spending private key
-   * @param viewingPrivateKey Receiver's viewing public key
-   * @param startBlock Block to start scanning from
-   * @param endBlock Block to scan until
+   * @notice Fetches all Umbra event logs between specified blocks using the standard RPC provider
+   * @param startBlock Block number to begin scanning for events
+   * @param endBlock Block number to end scanning for events, or 'latest'
+   * @returns A list of Announcement events supplemented with additional metadata, such as the sender, block,
+   * timestamp, and txhash
    */
-  async userAnnouncementsFromLogs(
-    spendingPublicKey: string,
-    viewingPrivateKey: string,
+  async fetchAllAnnouncementFromLogs(
     startBlock: string | number,
     endBlock: string | number
-  ): Promise<UserAnnouncement[]> {
+  ): Promise<AnnouncementDetail[]> {
+    // Graph query failed, try requesting logs directly IF we are not on polygon. If we are on
+    // Polygon, there isn't much we can do, so for now just show a warning and return an empty array
```

Comment:
two outdated comments here:
1. since this is a separate method, `Graph query failed` is outdated
2. comment says show a warning and return an empty array, but actually we throw an error

Reply:
fixed